WalletWasabi/WabiSabi/Backend/Statistics/RequestTimeStatista.cs

```diff
@@ -15,36 +15,53 @@ private RequestTimeStatista()
 
 	public static RequestTimeStatista Instance => Lazy.Value;
 
-	private Dictionary<string, List<(DateTimeOffset Time, TimeSpan Duration)>> Requests { get; } = new();
-	private object Lock { get; } = new();
+	private Dictionary<string, List<TimeSpanData>> TimeSpanDataList { get; } = [];
+	private Dictionary<string, List<IntegerData>> IntegerDataList { get; } = [];
+	private object TimeSpanDataLock { get; } = new();
+	private object IntegerDataLock { get; } = new();
 	private DateTimeOffset LastDisplayed { get; set; } = DateTimeOffset.UtcNow;
 	private TimeSpan DisplayFrequency { get; } = TimeSpan.FromMinutes(60);
 
 	public void Add(string request, TimeSpan duration)
 	{
 		try
 		{
-			var toDisplay = false;
-			lock (Lock)
+			lock (TimeSpanDataLock)
 			{
-				if (Requests.TryGetValue(request, out List<(DateTimeOffset Time, TimeSpan Duration)>? value))
+				TimeSpanData newItem = new(DateTimeOffset.UtcNow, duration);
+
+				if (TimeSpanDataList.TryGetValue(request, out List<TimeSpanData>? value))
 				{
-					value.Add((DateTimeOffset.UtcNow, duration));
+					value.Add(newItem);
 				}
 				else
 				{
-					Requests.Add(request, new() { (DateTimeOffset.UtcNow, duration) });
-				}
-
-				if (DateTimeOffset.UtcNow - LastDisplayed > DisplayFrequency)
-				{
-					toDisplay = true;
+					TimeSpanDataList.Add(request, [newItem]);
 				}
 			}
+		}
+		catch (Exception ex)
+		{
+			Logger.LogError(ex);
+		}
+	}
 
-			if (toDisplay)
+	public void Add(string request, int integer)
+	{
+		try
+		{
+			lock (IntegerDataLock)
 			{
-				Display();
+				IntegerData newItem = new(DateTimeOffset.UtcNow, integer);
```

Comment:
I don't get this `IntegerData` and new logging. 

What's the purpose of it?

We already log out how many times a method is called.
`RequestTimeStatista.Display (64)	Responded to 'connection-confirmation'	 5344 times. Median: 0.0s Average: 0.0s Largest 0.1s`

Reply:
This is for logging integer data. Like counters or like how many worker threads do we have. We need Min, Avg and Max for integers.